kernel_gateway/services/sessions/handlers.py

```diff
@@ -27,6 +27,23 @@ def get(self):
         else:
             super(SessionRootHandler, self).get()
 
+class SessionHandler(TokenAuthorizationMixin,
+                    CORSMixin,
+                    JSONErrorsMixin,
+                    notebook_handlers.SessionHandler):
+    """Extends the notebook session handler with token auth, CORS, and
+    JSON errors.
+    """
+    def set_default_headers(self):
+        self.set_header('Content-Type', 'application/json')
+        self.set_header('Access-Control-Allow-Origin', '*')
```

Comment:
I think this needs to be set based on the `allow_origin` trait: https://github.com/jupyter-server/kernel_gateway/blob/5269af0a2bf84b8a93b0c983d731b82843daec02/kernel_gateway/gatewayapp.py#L164

Reply:
@blink1073 - so something [like this](https://github.com/jupyter-server/kernel_gateway/blob/5269af0a2bf84b8a93b0c983d731b82843daec02/kernel_gateway/gatewayapp.py#LL484C9-L484C9)?
```suggestion
        self.set_header('Access-Control-Allow-Origin', self.settings['kg_allow_origin'])
```

Reply:
Yep, that should do it.

Reply:
@blink1073  - Hmm, sorry, but should [allow-methods and allow-headers be handled similarly](https://github.com/jupyter-server/kernel_gateway/blob/5269af0a2bf84b8a93b0c983d731b82843daec02/kernel_gateway/gatewayapp.py#L482-L483) - since these are configurable as well?

It's really not clear to me what this is all about.

Reply:
Yeah, we should use those as well.

Reply:
Thanks - those changes have been pushed.